### PR TITLE
Daemon mode with performance improvements

### PR DESCRIPTION
**Pull request by tptee**
Depends on https://github.com/FormidableLabs/inspectpack/pull/26

This PR introduces a child process daemon mode for interactive usage (e.g. `webpack-dashboard`). Creating a daemon prevents `inspectpack` from blocking UI interactions and allows for future parallelization of action runners.

## Performance improvements:
- Use `traverse.cheap` in the parser
- Remove all `gzipSync` in the `sizes` action for both daemon and CLI mode
- Add both an action-level and module-level in-memory + disk cache, making all builds except the first run cheaper

## Other improvements:
- Centralize file size gathering/caching into an injectable Compressor class

Todo:

- [x] PR/Fork `workerpool` to allow for worker args and inspector support (will fix compressor cache's support for custom `cacheDir`s)
- [x] Useful comments

### CODE REVIEW

**Comment by tptee** [comment]
Should be ready for a look!

File: lib/actions/base.js [hunk 1]
```diff
@@ -56,13 +56,13 @@ Base.prototype.display = function (data) {
...
     return this.tsvTemplate({
-      opts: opts,
+      opts,
       data: _.omit("meta")(data),
       meta: data.meta
     });
   default:
     // Programming error.
-    throw new Error("Unknown format: " + format);
+    throw new Error(`Unknown format: ${ format}`);
```
Comment (ryan-roemer) [inline lib/actions/base.js]: NIT: Space

File: lib/utils/hash.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+"use strict";
+
+const crypto = require("crypto");
+const tryRequire = require("try-require");
```
Comment (ryan-roemer) [inline lib/utils/hash.js]: I don't think we need a package to do:

```js
let farmhash;
try {
  farmhash = require("farmhash");
} catch (err) { /* passthrough * }
```

?

**Comment by tptee** [comment]
Released in `v1.1.0`!

**Pull request merged**